### PR TITLE
[express-slow-down] Add types

### PR DESCRIPTION
**Pull request by jdforsythe**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/express-slow-down/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        onLimitReached?(req: express.Request, res: express.Response, optionsUsed: Options): void;
+
+        /**
+         * The storage to use when persisting request attempts. By default, the MemoryStore is used.
+         */
+        store?: Store;
+    }
+}
+
+declare var SlowDown: new (options: SlowDown.Options) => express.RequestHandler;
```
Comment (sheetalkamat) [inline types/express-slow-down/index.d.ts]: https://github.com/nfriedly/express-slow-down#usage shows slowdown as function and not constructor?

File: types/express-slow-down/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+         * Function used to skip requests. Returning `true` from the function will skip delaying for that request.
+         * Default: `(req, res) => false`
+         */
+        skip?(req: express.Request, res: express.Response): boolean;
+
+        /**
+         * Function to execute the first time the limit is reached within `windowMs`.
+         * Default: `(req, res, opts) => {}`
+         */
+        onLimitReached?(req: express.Request, res: express.Response, optionsUsed: Options): void;
```
Comment (sheetalkamat) [inline types/express-slow-down/index.d.ts]: I think you also need to handle https://github.com/nfriedly/express-slow-down#reqslowdown

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

File: types/express-slow-down/index.d.ts (hunk 1 above)
Comment (jdforsythe) [reply types/express-slow-down/index.d.ts]: Sorry.. fixed in 45f8494

File: types/express-slow-down/index.d.ts (hunk 2 above)
Comment (jdforsythe) [reply types/express-slow-down/index.d.ts]: Handled in 45f8494

File: types/express-slow-down/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        /**
+         * Amount of delay imposed on current request in milliseconds
+         */
+        delay: number;
+    }
+
+    /**
+     * Express Request with the added `slowDown` property
+     */
+    type RequestWithSlowDown = express.Request & { slowDown: SlowDownRequestAugmentation }
```
Comment (sheetalkamat) [inline types/express-slow-down/index.d.ts]: I think this type can be
```ts
interface RequestWithSlowDown  extends express.Request {
slowDown: SlowDownRequestAugmentation;
}
```

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

File: types/express-slow-down/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+        onLimitReached?(req: RequestWithSlowDown, res: express.Response, optionsUsed: Options): void;
+
+        /**
+         * The storage to use when persisting request attempts. By default, the MemoryStore is used.
+         */
+        store?: Store;
+    }
+}
+
+declare var SlowDown: (options: SlowDown.Options) => express.RequestHandler;
```
Comment (sheetalkamat) [inline types/express-slow-down/index.d.ts]: This needs to be function instead of variable with function type

**Review by sheetalkamat — APPROVED** [review]

**Pull request merged**